tf2_ros/include/tf2_ros/message_filter.h

```diff
@@ -649,8 +648,23 @@ class MessageFilter : public MessageFilterBase, public message_filters::SimpleFi
     const MConstPtr & message = evt.getMessage();
     std::string frame_id = stripSlash(mt::FrameId<M>::value(*message));
     rclcpp::Time stamp = mt::TimeStamp<M>::value(*message);
-    RCLCPP_INFO(node_logging_->get_logger(), "[%s] Drop message: frame '%s' at time %.3f for reason(%d)",
-      __func__, frame_id.c_str(), stamp.seconds(), reason);
+
+    std::string reason_string;
```

Comment:
This logic would be great for a helper function that converts the enum to string to keep the complexity of the functions lower. It then could also have a small unit test to ensure coverage too.

Reply:
This function exists *just for logging*. It's appropriate to convert the enum here. Also, I have no idea what the enums actually mean beyond what their names mean.

Reply:
Just because it's used for logging doesn't mean it shouldn't have coverage and follow best practices.

The code would be much more readable if the whole case block was just a function `std::string get_failure_reason_string(const FilterFailureReason& reason)`

That then is invoked in the RCLCPP_INFO element instead of `reason_string.c_str()` it would be `get_failure_reason_string(reason).c_str()` And this method stays 5 lines long with no branchpoints. 

And a unit test can be written for the static function that exercises all 3 cases and provides full coverage of that function.

This is just cleaning up the structure of the code to avoid building up technical debt.